guides/source/action_view_overview.md

```diff
@@ -1505,6 +1505,266 @@ strip_tags("<b>Bold</b> no more!  <a href='more.html'>See more</a>")
 
 NB: The output may still contain unescaped '<', '>', '&' characters and confuse browsers.
 
+### UrlHelper
+Provides a set of methods for making links and getting URLs that
+depend on the routing subsystem (see ActionDispatch::Routing).
+This allows you to use the same format for links in views
+and controllers.
+
+#### url_for
+Resides in `ActionView::RoutingUrlFor` module.
+
+Returns the URL for the set of `options` provided. This takes the
+same options as `url_for` in Action Controller (see the
+documentation for `ActionController::Base#url_for`). 
+
+
+NOTE: By default
+`:only_path` is `true` so you'll get the relative `"/controller/action"`
+instead of the fully qualified URL like `http://example.com/controller/action`.
+
+##### Examples
+```ruby
+
+url_for(action: 'login', controller: 'members', only_path: false, protocol: 'https', anchor: 'tax&ship')
+# => https://www.example.com/members/login/#tax&ship
+
+url_for(@workshop)
+# calls @workshop.to_param which by default returns the id
+# => /workshops/5
+
+# to_param can be re-defined in a model to provide different URL names:
+# => /workshops/1-workshop-name
+ 
+url_for([@hotel, @booking, page: 2, line: 3])
+# => /hotels/1/bookings/1?line=3&page=2
+
+url_for(:back)
+# if request.env["HTTP_REFERER"] is set to "http://www.example.com"
+# => http://www.example.com
+
+```
+#### link_to
+ Creates an anchor element of the given `name` using a URL created by the set of `options`.
```

Comment:
I think ```Creates an anchor element of the given `name` using a URL created by the set of `options`.``` again with one example is preferred here.
```
link_to "Profile", profile_path(@profile)
# => <a href="/profiles/1">Profile</a>
```

Reply:
@gmcgibbon since `link_to` is highly useful so lets have 2 examples for it. I have reduced desc.